app/src/db/migrations/20240115201832_files-api-access.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function(knex) {
+  return Promise.resolve()
+    .then(() => knex.schema.alterTable('form_api_key', table => {
+      table.boolean('filesApiAccess').defaultTo(false).comment('Keeps track of whether files can be accessed using the API key');
+    }));
+};
```

Comment:
We do have at least one form that is calling the files route with the API Key. How do we deploy this change without breaking things for them? After the column is added, should we maybe set it to true for everyone, and they can disable if they want? Tricky!

This is the Kibana query I used to find the callers, where the existence of "formId" means that it's an API Key user, and that it passed authn/z.
```
kubernetes.namespace_labels.environment:prod AND kubernetes.container_name.raw:app AND message:"app/api/v1/files" AND message:"formId"
```

Reply:
I think setting it true for everyone would defeat the purpose of this feature, as it was added to create an additional check to limit access to files. Could we set it to true just for that one form? 

I haven't looked into the database yet but I believe it was only James who is using this feature so far, so I could let him know about this change to ensure that he turns the feature on manually in the required forms.

Reply:
I'm really worried about breaking existing functionality for our users. Even if someone has never made an API call with their key, we don't know if maybe they're creating a system that relies on the functionality. I think the safest thing would be to set all existing API Keys to allow the file access, and then ask users to turn it off if they don't need it. For all future API Keys that are created, agreed, the file access should be disabled by default.

The other approach is that we could contact every form owner who has an API Key and ask them if they use the file download or not. Perhaps that would be better? Maybe it's a Duncan/Bhuvan question?

Reply:
It's tough to say. I thought James was the only one who even knew about this feature but you are right, we don't know who's read the updated docs and what kind of systems they're creating. I like your plan, so we can merge this and as soon as it's deployed we manually set the filesApiAccess value to true for everyone.